src/catter/core/ipc.cc

```
@@ -3,12 +3,49 @@
+                                     BufferReader& buf_reader,
+                                     Writer&& writer) {
+        if constexpr(!std::is_same_v<Ret, void>) {
+            auto response =
+                Serde<Ret>::serialize(callback(Serde<Args>::deserialize(buf_reader)...));
+            auto ret = co_await writer(
+                merge_range_to_vector(Serde<size_t>::serialize(response.size()), response));
+            if(ret.has_error()) {
+                throw std::runtime_error(std::format("Failed to send response [{}] to client: {}",
+                                                     eventide::refl::enum_name<Req>(),
+                                                     ret.message()));
+            }
```

Cppcheck (2.19.0)

[error] Found an exit path from function with non-void return type that has missing return statement

(missingReturn)